src/Classifiers/OneVsRest.php

```diff
@@ -0,0 +1,254 @@
+     * A map of each class to its binary classifier.
+     *
+     * @var array<\Rubix\ML\Learner>
+     */
+    protected array $classifiers = [
+        //
+    ];
+
+    /**
+     * The dimensionality of the training set.
+     *
+     * @var int<0,max>|null
+     */
+    protected ?int $featureCount = null;
+
+    /**
+     * @param \Rubix\ML\Learner $base
+     * @throws \Rubix\ML\Exceptions\InvalidArgumentException
+     */
+    public function __construct(Learner $base)
+    {
+        if (!$base->type()->isClassifier()) {
+            throw new InvalidArgumentException('Base Learner must be'
+                . ' a classifier.');
+        }
+
+        if (!$base instanceof Probabilistic) {
+            throw new InvalidArgumentException('Base classifier must'
+                . ' implement the Probabilistic interface.');
+        }
+
+        $this->base = $base;
+        $this->backend = new Serial();
+    }
+
+    /**
+     * Return the estimator type.
+     *
+     * @internal
+     *
+     * @return \Rubix\ML\EstimatorType
+     */
+    public function type() : EstimatorType
+    {
+        return EstimatorType::classifier();
+    }
+
+    /**
+     * Return the data types that the estimator is compatible with.
+     *
+     * @internal
+     *
+     * @return list<\Rubix\ML\DataType>
+     */
+    public function compatibility() : array
+    {
+        return $this->base->compatibility();
+    }
+
+    /**
+     * Return the settings of the hyper-parameters in an associative array.
+     *
+     * @internal
+     *
+     * @return mixed[]
+     */
+    public function params() : array
+    {
+        return [
+            'base' => $this->base,
+        ];
+    }
+
+    /**
+     * Has the learner been trained?
+     *
+     * @return bool
+     */
+    public function trained() : bool
+    {
+        return !empty($this->classifiers);
+    }
+
+    /**
+     * Train the learner with a dataset.
+     *
+     * @param \Rubix\ML\Datasets\Labeled $dataset
+     */
+    public function train(Dataset $dataset) : void
+    {
+        SpecificationChain::with([
+            new DatasetIsLabeled($dataset),
+            new DatasetIsNotEmpty($dataset),
+            new SamplesAreCompatibleWithEstimator($dataset, $this),
+            new LabelsAreCompatibleWithLearner($dataset, $this),
+        ])->check();
+
+        $classes = $dataset->possibleOutcomes();
+
+        $this->backend->flush();
+
+        foreach ($classes as $class) {
+            $estimator = clone $this->base;
+            $subset = clone $dataset;
+
+            $binarize = function ($label) use ($class) {
+                return $label === $class ? 'y' : 'n';
+            };
+
+            $subset->transformLabels($binarize);
+
+            $task = new TrainLearner($estimator, $subset);
+
+            $this->backend->enqueue($task);
+        }
+
+        $classifiers = $this->backend->process();
+
+        $classifiers = array_combine($classes, $classifiers) ?: [];
```

Comment:
```suggestion
        $classifiers = array_combine($classes, $classifiers);
```

Reply:
Unfortunately, this is needed for PHP 7.4 :/